App/src/main/java/cc/softwarefactory/lokki/android/fragments/MapViewFragment.java

```diff
@@ -592,32 +558,17 @@ protected void onPostExecute(Bitmap bitmapResult) {
             if (bitmapResult == null || cancelAsyncTasks || !isAdded() || map == null) {
                 return;
             }
-            Person marker = markerMap.get(email);
-
-            Boolean isNew = false;
-            if (marker != null) {
-                Log.d(TAG, "onPostExecute - updating marker: " + email);
-                markerMap.remove(marker);
-                clusterManager.removeItem(marker);
-                Person p = new Person(latLng, email, time, bitmapResult);
-                markerMap.put(email, p);
-                clusterManager.addItem(p);
-            } else {
-                Log.d(TAG, "onPostExecute - creating marker: " + email);
-                marker = new Person(latLng, email, time, bitmapResult);
+            Person marker = person;
+
+            if (!markers.contains(marker)) {
+                markers.add(marker);
```

Comment:
You have to remove marker from clustermanager and add it again, as there is no way to update cluster